cabal-install/Distribution/Client/Dependency/Types.hs

```diff
@@ -105,6 +106,9 @@ instance Text PreSolver where
       "choose"  -> return Choose
       _         -> Parse.pfail
 
+newtype EnableBackjumping = EnableBackjumping Bool
+  deriving Show
```

Comment:
I think it's good that you introduce a `newtype` here. We should probably do the same for all the other `Bool` arguments to the solver. Would you perhaps be willing to do this (in a separate PR)?